oggm/core/inversion.py

```diff
@@ -174,29 +291,43 @@ def mass_conservation_inversion(gdir, glen_a=cfg.A, fs=0., write=True,
 
         # Parabolic bed rock
         w = cl['width']
-        a0s = - cl['flux_a0'] / ((cfg.RHO*cfg.G*slope)**3*fd)
 
-        if np.any(~np.isfinite(a0s)):
-            raise RuntimeError('({}) something went wrong with the '
-                               'inversion'.format(gdir.rgi_id))
+        a0s = - cl['flux_a0'] / ((cfg.RHO*cfg.G*slope)**3*fd)
 
-        # GO
-        out_thick = np.zeros(len(slope))
-        for i, (a0, Q) in enumerate(zip(a0s, cl['flux_a0'])):
-            if Q > 0.:
-                out_thick[i] = _inv_function(a3, a0)
-            else:
-                out_thick[i] = 0.
-        assert np.all(np.isfinite(out_thick))
+        # Start iteration for shape factor with guess of 1
+        i = 0
+        sf = np.ones(slope.shape)
+        sf_diff = np.ones(slope.shape)
```

Comment:
``i = 0`` and ``sf_diff`` could be defined in the ``if`` block below

Reply:
Absolutely right